website/client/js/controllers/settingsCtrl.js

```diff
@@ -287,6 +293,37 @@ habitrpg.controller('SettingsCtrl',
       return Math.floor(numberOfHourglasses);
     };
 
+    $scope.hasSocialAuth = function(user) {
+      for (var id in SOCIAL_AUTH_NETWORKS) {
+        var network = SOCIAL_AUTH_NETWORKS[id];
+        if (user.auth.hasOwnProperty(network.key)) {
+          if (user.auth[network.key].id) {
+            return true;
+          }
+        }
+      }
+      return false;
+    };
+
+    $scope.hasBackupAuthOption = function(user, checkedNetworkKey) {
+      if (user.auth.local.username) {
+        return true;
+      }
+      for (var id in SOCIAL_AUTH_NETWORKS) {
```

Comment:
Same as above. A `forEach` should be used. `for in`s can have weird behavior if a browser extension has added something to the object or array prototypes.